awscli/customizations/history/list.py

```diff
@@ -0,0 +1,456 @@
+        'Each row shows only a bare minimum of details including the '
+        'command_id, date, arguments and return code. You can use the '
+        '``history show`` with the command_id to see more details about '
+        'a particular entry.'
+    )
+    _COL_WIDTHS = OrderedDict([
+        ('Command Id', 38),
+        ('Time', 24),
+        ('Arguments', 50),
+        ('Return Code', 13)
+    ])
+
+    def _run_main(self, parsed_args, parsed_globals):
+        self._connect_to_history_db()
+        raw_records = self._db_reader.iter_all_records()
+        records = RecordAdapter(raw_records)
+
+        with self._get_output_stream('less -SR') as output_stream:
+            use_color = self._should_use_color(parsed_globals)
+            styler = TableStyler(output_stream, colorize=use_color)
+            formatter = TableFormatter(self._COL_WIDTHS, styler)
+            formatter('history list', records)
+        return 0
+
+
+class RecordAdapter(object):
+    def __init__(self, records):
+        self._records = records
+
+    def __iter__(self):
+        for record in self._records:
+            adapted_record = self._format_record(record)
+            yield adapted_record
+
+    def _format_time(self, timestamp):
+        command_time = datetime.datetime.fromtimestamp(timestamp / 1000)
+        formatted = datetime.datetime.strftime(
+            command_time, '%Y-%m-%d %I:%M:%S %p')
+        return formatted
+
+    def _format_args(self, args):
+        json_value = json.loads(args)
+        formatted = json_value
+        return formatted
+
+    def _format_record(self, record):
+        formatted = OrderedDict([
+            ('Command Id', record['id_a']),
+            ('Time', self._format_time(record['timestamp'])),
+            ('Arguments', self._format_args(record['args'])),
+            ('Return Code', record['rc'])
+        ])
+        return formatted
+
+
+class TableCellFormatter(object):
+    """Class for splitting table cell data across multiple rows
+
+    Each table row is represented by a record from the database. Each element
+    of that record represents an individual table cell's value. For the three
+    columns Command Id, Time, and Return Code the contents will fit within
+    their table cell since they are fixed in width.
+
+    The Arguments table cell value can be arbitrarily wide since it is a
+    command entered by the user. In order to fit it into a table cell it must
+    be split into a number of terminal rows.
+    This class is responsible for wrapping a table cell value and providing an
+    interface for asking for just 1 terminal row to write.
+
+    Now when rendering a table row, each column can be iterated over asking
+    its TableCellFormatter if it has a terminal row to write. This is repeated
+    until all TableCellFormatters indicate that they are out of content, then
+    the table row can be capped with a footer.
+    """
+    def __init__(self, value, space):
```

Comment:
So I still am not following what the `value` and `space` are for. Could we add docstrings describing these in both the `__init__()` and `get_line(self, space)`?